.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,77 @@
+name: CI/CD
+
+on:
+  push:
+    branches: [dev, master]
+    tags:
+      - v**
+  pull_request:
+
+jobs:
+  validate:
+    name: API Docs Validation
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: swagger-validator
+        uses: mbowman100/swagger-validator-action@2.0
+        with:
+          files: |
+            swagger.yml
+            master-runner/swagger.yml
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: "11"
+          distribution: "adopt"
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@e6e38bacfdf1a337459f332974bb2327a31aaf4b
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+      - name: Run Gradle Check
+        run: ./gradlew check dist
+      - name: Build with Gradle
+        run: ./gradlew dist
+      - name: Login to Dockerhub
+        if: |
+          contains(github.ref, 'master') ||
+          contains(github.ref, 'dev') ||
+          startsWith(github.ref, 'refs/tags/v')
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_LOGIN }}
+          password: ${{ secrets.DOCKER_PWD }}
+      - name: Deploy to Dockerhub
```

Comment:
All Docker actions within one script, either to be called from here or within this workflow.

Reply:
Yes, that was my plan too, but I always get this error when I try to login to Dockerhub in Github Actions with `echo $DOCKER_PWD | docker login -u $DOCKER_LOGIN --password-stdin`:

```
Error: Cannot perform an interactive login from a non TTY device
```

Reply:
You could try:

```bash
docker login -u $DOCKER_LOGIN -p $DOCKER_PWD
```

(Note that this has some negative security implications, but I don't think these are relevant in this case)

Reply:
> You could try:
> 
> ```shell
> docker login -u $DOCKER_LOGIN -p $DOCKER_PWD
> ```
> 
> (Note that this has some negative security implications, but I don't think these are relevant in this case)


The same error occurs with these

Reply:
It seems that in GitHub Actions, secrets have to be explicitly set as environment variables:

```yaml
env:
  DOCKER_LOGIN: ${{ secrets.DOCKER_LOGIN }}
  DOCKER_PWD: ${{ secrets.DOCKER_PWD }}
```

This seams to be missing, at least in the currently pushed version. If you didn't already, tried that. This might be the problem.

Reply:
> It seems that in GitHub Actions, secrets have to be explicitly set as environment variables:
> 
> ```yaml
> env:
>   DOCKER_LOGIN: ${{ secrets.DOCKER_LOGIN }}
>   DOCKER_PWD: ${{ secrets.DOCKER_PWD }}
> ```
> 
> This seams to be missing, at least in the currently pushed version. If you didn't already, tried that. This might be the problem.

Yes, that just occurred to me too :smile: